docs/source/data/errors.mdx

```diff
@@ -332,130 +346,135 @@ You can edit Apollo Server error details before they're passed to a client or re
+  }
+  // highlight-end
+
+  // Otherwise return the formatted error.
+  return formattedError;
+},
 ```
 
-> To make context-specific adjustments to the error received by `formatError` (such as localization or personalization), consider [creating a plugin](../integrations/plugins/) that uses the [`didEncounterErrors` lifecycle event](../integrations/plugins-event-reference/#didencountererrors) to attach additional properties to the error. These properties can be accessed from `formatError`.
+> To make context-specific adjustments to the error received by `formatError` (such as localization or personalization), consider [creating a plugin](/apollo-server/integrations/plugins/) that uses the [`didEncounterErrors` lifecycle event](/apollo-server/integrations/plugins-event-reference/#didencountererrors) to attach additional properties to the error. These properties can be accessed from `formatError`.
 
 ### For Apollo Studio reporting
 
 You can use Apollo Studio to analyze your server's error rates. If you connect Apollo Server to Studio, all errors are sent to Studio by default. If you _don't_ want certain error information to be sent to Studio (either because the error is unimportant or because certain information is confidential), you can modify or redact errors entirely before they're transmitted.
 
-To accomplish this, you can provide a `rewriteError` function to the [usage reporting plugin](../api/plugin/usage-reporting/).
+To accomplish this, you can provide a `rewriteError` function to the [usage reporting plugin](/apollo-server/api/plugin/usage-reporting/).
 
 > The usage reporting plugin is installed automatically with its default configuration if you provide an Apollo API key to Apollo Server. To define a custom `rewriteError` function, you need to install the plugin explicitly with a custom configuration, as shown in examples below.
 
-Your `rewriteError` function is called for each error (a `GraphQLError` or an `ApolloError`) to be reported to Studio. The error is provided as the function's first argument. The function can either:
+Your `rewriteError` function is called for each error (`GraphQLError`) to be reported to Studio. The error is provided as the function's first argument. The function can either:
 
-* Return a modified form of the error (e.g., by changing the `err.message` to remove potentially sensitive information)
-* Return `null` to prevent the error from being reported entirely
+- Return a modified form of the error (e.g., by changing the `err.message` to remove potentially sensitive information)
+- Return `null` to prevent the error from being reported entirely
 
->**For federated graphs**, instead define `rewriteError` in each subgraph's [inline trace plugin](../api/plugin/inline-trace/#rewriteerror). Do not define it in the gateway.
+> **For federated graphs**, instead define `rewriteError` in each subgraph's [inline trace plugin](/apollo-server/api/plugin/inline-trace/#rewriteerror). Do not define it in the gateway.
```

Comment:
slight nuance (not new, just noticing it now): you can define it in the gateway if you're trying to rewrite stuff like parse and validation errors that happen in the gateway. it just won't see field errors.

Reply:
Cool! Re wrote this as:

> **For federated graphs**, define your `rewriteError` function in each subgraph's [inline trace plugin](./migration/rewriteEerror-plugin-option) to rewrite field errors. If you want to transform your gateway's parsing or validation errors, you can define your `rewriteError` function in your gateway.